urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/group/Group.java

```diff
@@ -0,0 +1,128 @@
+package com.sinopec.group;
+
+import com.sinopec.bean.AdjacentGroup;
+import com.sinopec.cache.InstantMsgCache;
+import com.sinopec.cache.HistroyMsgCache;
+import com.sinopec.common.Constants;
+import com.sinopec.group.rpc.DynamicInfoServer;
+import com.sinopec.group.rpc.TopologyClient;
+import com.sinopec.jdbc.DBOpt;
+import com.sinopec.jdbc.JDBCCommon;
+import com.sinopec.utils.ConfigUtils;
+import com.sinopec.utils.HostUtil;
+import com.sinopec.utils.LogUtils;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Created with IntelliJ IDEA. User: Administrator Date: 13-10-23 Time: 下午3:50
+ * To change this template use File | Settings | File Templates.
+ */
+public class Group {
+
+	private static final Log LOG = LogFactory.getLog(Group.class);
+
+	public void startDynamicInfoListen(int port, InstantMsgCache dynamicCache, HistroyMsgCache histroyMsgCache, AdjacentGroup adjacentGroup) {
+		new Thread(new DynamicInfoServer(port, dynamicCache, histroyMsgCache, adjacentGroup)).start();
+	}
+
+	public void startThreadAsClient(String crontrolNodeIP, int port, int samplingInterval, String localhostIP, AdjacentGroup adjacentGroup) {
+		new Thread(new TopologyClient(crontrolNodeIP, port, samplingInterval * 1000, localhostIP, adjacentGroup)).start();
+	}
+
+	private ConfigUtils getConfigUtils(String filePath) {
+		return new ConfigUtils(filePath);
+	}
+
+	private boolean isMySQL(String dbType) {
+		if (dbType.equalsIgnoreCase("mysql")) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	private boolean isOracle(String dbType) {
+		if (dbType.equalsIgnoreCase("Oracle")) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	public void start() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 50 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 68-121 into a separate method.